dev/test_e2e.py

```diff
@@ -0,0 +1,336 @@
+
+
+def list_some_cases(mongo: object, country: str) -> None:
+	try:
+		records = list(mongo.find({"location.country": country}))
+		mid_index = len(records) // 2
+		for _ in range(mid_index):
+			mongo.update_one({"location.country": country}, {"$set": {"list": True}})
+	except Exception as exc:
+		raise Exception(f"An exception occurred while trying to add `list:true` to cases: {exc}")
+
+
+class TestE2E():
+
+	@classmethod
+	def setup_class(cls):
+		check_env()
+		with webdriver.Firefox(options=SELENIUM_OPTIONS) as driver:
+			sign_up(driver)
+
+	@pytest.mark.skip("Not all ingestors working locally")		
+	def test_all_cases_ingested(self, mongo):
+		source_data_files = []
+		response = requests.get(f"{MOCK_SOURCE_DATA_ADDRESS}/sources")
+
+		sources = response.json()
+		for _, val in sources.items():
+			if s3_object_location := val.get("origin", {}).get("url"):
+				source_data_files.append(s3_object_location)
+
+		files_case_count = 0
+		for source_file in source_data_files:
+			file_name = source_file.split("/")[-1]
+			cases = get_cases(file_name)
+			files_case_count += len(cases)
+
+		db_case_count = 0
+		try:
+			db_case_count = mongo.count_documents({})
+		except Exception as exc:
+			raise Exception(f"An exception occurred while trying to count case documents: {exc}")
+
+		assert files_case_count == db_case_count
+
+	@pytest.mark.parametrize("country", get_countries())
+	def test_cases_in_db(self, mongo, country):
+		# TODO: update this for countries with multiple sources
+		records = get_country_records(mongo, country)
+		response = requests.get(f"{MOCK_SOURCE_DATA_ADDRESS}/names")
+		response = requests.get(f"{MOCK_SOURCE_DATA_ADDRESS}/names/{country.lower()}")
+		source_id = response.text
+		response = requests.get(f"{MOCK_SOURCE_DATA_ADDRESS}/sources/{source_id}")
+		source_details = response.json()
+		source_file = source_details.get("origin", {}).get("url")
+		file_name = source_file.split("/")[-1]
+		case_data = get_cases(file_name)
+		cases = []
+		if country == "Cuba":
+			cases = extract_cuba_cases(case_data)
+		else:
+			cases = case_data
+		assert len(records) == len(cases)
+
+	@pytest.mark.parametrize("country", get_countries())
+	def test_cases_geocoded(self, mongo, country):
+		records = []
+		try:
+			records = list(mongo.find({"country": country}))
+		except Exception as exc:
+			raise Exception(f"An exception occurred while trying to query the database: {exc}")
+		for record in records:
+			assert record.get("location")
+			assert record.get("location", {}).get("country")
+			assert record.get("location", {}).get("geometry")
+
+	@pytest.mark.parametrize("country", get_countries())
+	def test_listed_cases_show_on_ui(self, mongo, country):
+		# Cases in database with list:true should show in UI
+		# Cases in database without list:true should not show in UI
+		list_some_cases(mongo, country)
+		db_case_ids = []
+		try:
+			records = list(mongo.find({"location.country": country, "list": True}))
```

Comment:
Better to use collation (en_US, strength 2) to prevent the issue in #2298

Reply:
Done